torchvision/transforms/v2/functional/_color.py

```diff
@@ -56,12 +47,19 @@ def _rgb_to_grayscale_image_tensor(
     return l_img
 
 
+@_register_kernel_internal(rgb_to_grayscale, torch.Tensor)
 @_register_kernel_internal(rgb_to_grayscale, datapoints.Image)
 def rgb_to_grayscale_image_tensor(image: torch.Tensor, num_output_channels: int = 1) -> torch.Tensor:
+    if num_output_channels not in (1, 3):
```

Comment:
For some reason, this was done on the dispatcher rather than in the kernel. Same for the PIL kernel below.